### DISCUSSION TITLE
Good docs for code

### DISCUSSION THREAD

**Discussion by Tusenka** [prometheus/prometheus#15126] [Ideas]
Is there any good docs for code. I really messad up with this(

**Comment by beorn7** [comment]
It would indeed be very helpful to have a (more or less) high-level "tour of the codebase" somewhere.

However, it would be a lot of work, and it also needed to be maintained to not lose value very quickly.

While I'm sure anyone volunteering to do this would be very welcome, I assume in practice we have to rely on good doc comments (which can be browsed here in rendered form: https://pkg.go.dev/github.com/prometheus/prometheus ). And those are often quite lagging. At least there is a dev-summit agenda to discuss the state of doc comments.

**Reply by Tusenka** [reply]
Thank you!

**Comment by beorn7** [comment]
https://ganeshvernekar.com/blog/prometheus-tsdb-the-head-block/ is the start of a blog post series about TSDB, which dives a bit into code organization (for TSDB only).

**Reply by Tusenka** [reply]
Thank you!
